### PR TITLE
db/node: Make block_no NULL-able for EBBs and the fake genesis block

### PR DESCRIPTION
**Pull request by erikd**
And fix the knock on effects.

### CODE REVIEW

File: cardano-explorer-db/src/Explorer/DB/Query.hs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
```
Comment (cleverca22) [inline cardano-explorer-db/src/Explorer/DB/Query.hs]: what is `RankNTypes` doing in this context?

File: cardano-explorer/src/Explorer/Web/ClientTypes.hs [hunk 2]
```diff
@@ -116,7 +116,7 @@ adaToCCoin (Ada ada) = CCoin $ T.pack $ showFixed True $ ada * 1000000
 data CBlockEntry = CBlockEntry
     { cbeEpoch      :: !Word64
     , cbeSlot       :: !Word16
-    , cbeBlkHeight  :: !Word
+    , cbeBlkHeight  :: !(Maybe Word)
```
Comment (cleverca22) [inline cardano-explorer/src/Explorer/Web/ClientTypes.hs]: changing the type in here will break the compatibility with the old explorer clients, would be better to use `maybe` or `fromMaybe` to convert it down

the api should return a `Left` for EBB's, so they never have to be represented by a `CBlockEntry`

**Review by cleverca22 — CHANGES_REQUESTED** [review]

File: cardano-explorer-db/src/Explorer/DB/Query.hs (hunk 1 above)
Comment (erikd) [reply cardano-explorer-db/src/Explorer/DB/Query.hs]: It was needed for a something I was trying and then removed.

Have now removed it.

File: cardano-explorer/src/Explorer/Web/ClientTypes.hs (hunk 2 above)
Comment (erikd) [reply cardano-explorer/src/Explorer/Web/ClientTypes.hs]: Yeah, I can use a `fromMaybe`

**Review by cleverca22 — APPROVED** [review]

**Pull request merged**